src/Products/SiteErrorLog/tests/testInitialization.py

```diff
@@ -0,0 +1,83 @@
+##############################################################################
+#
+# Copyright (c) 2001, 2002 Zope Foundation and Contributors.
+# All Rights Reserved.
+#
+# This software is subject to the provisions of the Zope Public License,
+# Version 2.1 (ZPL).  A copy of the ZPL should accompany this distribution.
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
+# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+
+import os
+import shutil
+import tempfile
+import unittest
+
+from App.config import getConfiguration, setConfiguration
+from OFS.Application import Application, AppInitializer
+from Zope2.Startup.options import ZopeWSGIOptions
+
+good_cfg = """
+instancehome <<INSTANCE_HOME>>
+
+<zodb_db main>
+   mount-point /
+   <mappingstorage>
+      name mappingstorage
+   </mappingstorage>
+</zodb_db>
+"""
+
+original_config = None
+
+
+def getApp():
+    from App.ZApplication import ZApplicationWrapper
+    DB = getConfiguration().dbtab.getDatabase('/')
+    return ZApplicationWrapper(DB, 'Application', Application)()
+
+
+class TestInitialization(unittest.TestCase):
+    """ Test the application initialization """
+
+    def setUp(self):
+        global original_config
```

Comment:
Why do you use a `global` variable here? As the original config is restored in `tearDown` it could be stored on `self`. This allows to get rid of the `if`. I do not expect that `getConfiguration()` is that expensive so it can be called only once for the whole test class.